##########################################################################
#
# IMPORTANT: any changes to property keys made in this file
#            REQUIRES the same changes in the MessageKeys class.
#            See: messages.com.wynd.vop.framework.MessageKeys.java
#
##########################################################################

# no args
NO_KEY=NO_KEY
# {0} = warning message
WARN={0}
# {0} = warning message
PROPAGATE={0}

# {0} = class simple name
vop.framework.dev.illegal.instantiation={0} is a class for statics. Do not instantiate it.
# {0} = class being instantiated; {1} = action being taken; {2} = type being acted against; {3} super-interface
vop.framework.dev.illegal.invocation=Could not find or instantiate class {0} to {1} given object of type {2}. Ensure that it has a no-arg constructor, and implements {3}.

# {0} is the name of the class being registered as a bean; {1} is the ObjectName being registered
vop.jmx.cache.naming.malformed={0} could not be registered on the MBeanServer because the name [{1}] is malformed.
# {0} is the class name or JMX ObjectName being registered / deregistered
vop.jmx.registration.pre=Problem with pre-registration or pre-deregistration of JMX MBean {0}.
# {0} is the class name or JMX ObjectName
vop.jmx.bean.noncompliant={0} is not a JMX compliant MBean.

# Last resort, unexpected exception; {0} = exception class simple name; {1} = exception message
vop.framework.global.general.exception=Unexpected exception {0}: {1}
# {0} = class name
vop.framework.exception.handler.error.values=Could not instantiate VopRuntimeException using throwable {0}.
# {0} = class name
vop.framework.exception.handler.error.cast=Could not cast {0} to VopRuntimeException.
# MethodArgumentNotValidException; {0} = "field" or "object"; {1} = codes; {2} = default message
vop.framework.global.validator.method.argument.not.valid=Argument not valid, {0} error: {1} - {2}
# HttpClientErrorException; {0} = reason phrase; {1} = exception message
vop.framework.global.http.client.error=Client Error {0}: {1}
# MethodArgumentTypeMismatchException; {0} = argument name; {1} = expected class name
vop.framework.global.rest.api.type.mismatch={0} should be of type {1}.
# ConstraintViolationException; {0} = bean class name; {1} = property name; {2} = violation message
vop.framework.global.validator.constraint.violation={0} {1}: {2}

# JAXB Marshaller configuration failed; no args
vop.framework.rest.config.marshaller.failed=Error configuring JAXB marshaller. Please investigate.
# JAXB Marshaller configuration failed; no args
vop.framework.rest.config.webservice.template.failed=Unexpected exception thrown by WebServiceTemplate. Please investigate.
# Propogate message from other service; {0} = message key;  {1} = message text
vop.framework.feign.message.received=External Service returned {0}: {1}

# no args
vop.framework.security.token.blank=No JWT Token in Header.
# no args
vop.framework.security.token.invalid=Invalid Token.
# {0} = the problem; {1} = the token; {2} = the simple class name of the exception {3} = message from the exception
vop.framework.security.token.broken={0}[{1}]; {2}[{3}]
# no args
vop.framework.security.traits.correlationid.blank=Cannot process blank correlation id.
# {0} = min elements (currently 2 to 5)
vop.framework.security.traits.correlationid.invalid=Invalid number of elements in correlation id, should be between {0} and {1} elements.
# {0} = IdTypes.[value]
vop.framework.security.traits.idtype.invalid=IdType {0} does not exist.
# {0} = Issuers.[value]
vop.framework.security.traits.issuer.invalid=Issuer {0} does not exist.
# {0} = Sources.[value]
vop.framework.security.traits.source.invalid=Source {0} does not exist.
# {0} = UserStatus.[value]
vop.framework.security.traits.userstatus.invalid=UserStatus {0} does not exist.
# {0} = kind of object that was being encrypted
vop.framework.security.encrypt.failed=Failed to encrypt {0}.
# {0} = kind of object that was being signed
vop.framework.security.sign.failed=Failed to sign {0}.
# {0} = action being taken on the attribute; {1} attribute name
vop.framework.security.attribute.failed=Could not {0} {1} attribute.
# no args
vop.framework.security.saml.insert.failed=Error while attempting to insert SAML Assertion into message.
# {0} = exception simple class name; {1} = exception message
vop.framework.security.ssl.context.failed=Could not establish SSL context due to {0}: {1}.
# {0} = operation that failed
vop.framework.security.sanitize.failed=Unexpected error in {0}.

# {0} = advice name, {1} = operation attempted
vop.framework.audit.cache.error.unexpected={0} Exception occurred while attempting to {1}.
# {0} = advice name, {1} = operation attempted
vop.framework.audit.aspect.error.unexpected={0} Exception occurred while attempting to {1}.
# {0} = advice name, {1} = operation attempted
vop.framework.audit.aspect.error.cannot.audit={0} - Throwable occurred while attempting to writeAuditError for {1}.

# no args
vop.framework.validator.initialize.error.unexpected=An unexpected error occurred while initializing validation.
# {0} = assertion message
vop.framework.validator.assertion={0}
# {0} the object that cannot be null
vop.framework.validator.not.null={0} cannot be null.
# {0} = validated object class name; {1} = expected class name
vop.framework.validator.type.mismatch={0} is not of type {1}.

# Simulator could not find mock response file; {0} = XML file name; {1} = key used to construct file name
vop.framework.remote.mock.not.found=Could not read mock XML file {0} using key {1}. Please make sure this response file exists in the main/resources directory.
# RemoteServiceCallMock is not set up to process a type; {0} = the RemoteServiceCallMock class; {1} = the class used in the request
vop.framework.remote.mock.unknown.type={0}.getKeyForMockResponse(..) does not have a file naming block for requests of type {1}

# no args
vop.framework.sqs.services.transfer.error=An exception has occurred while trying to send a message to SQS.
# {0} = exception message
vop.framework.sqs.services.transfer.message.error=An exception has occurred while trying to send a message to SQS. {0}
# no args
vop.framework.sqs.services.transfer.message.id.null=An exception has occurred while trying to send a message to SQS - Message ID cannot be null after sent message

# no args
vop.framework.sqs.services.create.error=An exception has occurred while trying to create a text message.
# {0} = exception message
vop.framework.sqs.services.create.message.error=An exception has occurred while trying to create a text message. {0}
# no args
vop.framework.sqs.services.create.jms.error=An JMS exception has occurred while trying to create a text message.
# {0} = exception message
vop.framework.sqs.services.create.jms.message.error=An JMS exception has occurred while trying to create a text message. {0}
